net/host.go

```diff
@@ -133,6 +151,9 @@ func (h *Host) CloseProtocolStream(id types.Hash) {
 
 // Advertise advertises in the DHT.
 func (h *Host) Advertise(strs []string) {
+	if h.isRelayer {
+		strs = append(strs, relayerProvidesStr)
```

Comment:
We don't want the offer manager overwriting our desire to advertise that we are a relayer, and it doesn't seem like the offer manager should need to know that we are a relayer, so I had to put this hack in. We'll have to give more thought to these APIs.